packages/devtools_app/lib/src/framework/notifications_view.dart

```diff
@@ -40,53 +38,12 @@ class NotificationsView extends StatelessWidget {
   }
 }
 
-class NotificationController implements NotificationService {
-  final toPush = ValueNotifier<NotificationMessage>(NotificationMessage(''));
-  final toDismiss = ValueNotifier<NotificationMessage>(NotificationMessage(''));
-
-  @visibleForTesting
-  final inProcess = <NotificationMessage>[];
-
-  /// Pushes a notification [message].
-  @override
-  bool push(String message) => smartPush(NotificationMessage(message));
-
-  /// Pushes a notification [message].
-  ///
-  /// Ignores the message if [allowDuplicates] is false and a message with the
-  /// same text is currently displayed to the user.
-  @override
-  bool smartPush(
-    NotificationMessage message, {
-    bool allowDuplicates = true,
-  }) {
-    if (!allowDuplicates &&
-        inProcess.containsWhere((m) => m.text == message.text)) {
-      return false;
-    }
-    inProcess.add(message);
-    toPush.value = message;
-    return true;
-  }
-
-  /// Dismisses all notifications with a matching message.
-  @override
-  void dismiss(String message) =>
-      toDismiss.value = NotificationMessage(message);
-
-  /// Marks the message as complete, so that the messages not
-  /// allowing duplicates,
-  /// with the same text, do not get rejected.
-  void markComplete(NotificationMessage message) {
-    inProcess.removeWhere((element) => element == message);
-  }
-
-  void dispose() {
-    toPush.dispose();
-    toDismiss.dispose();
-  }
-}
-
+// TODO(polina-c): it seems it would be better to combine the NotificationsView
+//  and the _Notifications widget since this one is so minimal and that would
+//  eliminate some unnecessary boilerplate. Especially since the
+//  _NotificationsState build method just returns return widget.child.
+// However, the direct merge of these classes does not work for some reasons.
```

Comment:
Oh I see the issue. We are calling `Overlay.of(context)` from lifecycle methods in `_NotificationsState`, which would fail if the `Overlay` widget is defined in `_NotificationsState.build` because there would be no Overlay in the tree at the time `Overlay.of(context)` is called. You can disregard then.

Reply:
added this to the comment